pkg/virt-launcher/virtwrap/converter/compute/input_device.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package compute
+
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type InputDeviceDomainConfigurator struct {
+	architecture string
+}
+
+func NewInputDeviceDomainConfigurator(architecture string) InputDeviceDomainConfigurator {
+	return InputDeviceDomainConfigurator{
+		architecture: architecture,
+	}
+}
+
+func (i InputDeviceDomainConfigurator) Configure(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
+	if vmi.Spec.Domain.Devices.Inputs != nil {
+		for idx := range vmi.Spec.Domain.Devices.Inputs {
+			inputDevice := api.Input{}
+			if err := convertInputDevice(&vmi.Spec.Domain.Devices.Inputs[idx], &inputDevice); err != nil {
+				return err
+			}
+			domain.Spec.Devices.Inputs = append(domain.Spec.Devices.Inputs, inputDevice)
+		}
+	}
```

Comment:
This looks cleaner, but it introduces a subtle side effect. Because we are appending directly inside the loop, an error would leave `domain.Spec.Devices.Inputs` partially populated (whereas the original only updated it on success).

To be safe, could we please move the original code as-is for this step? We can refactor it in a follow-up commit / PR.